core/cache/watcher.go

```diff
@@ -258,3 +259,37 @@ func (w *ChangeWatcher) changed(topic string, value interface{}) {
 
 	w.notify(strings)
 }
+
+// RegexpChangeWatcher notifies when individual pieces of a subscribed topic
+// match the given compiled regular expression.  An initial event is sent
+// with the input given at creation.
+type RegexpChangeWatcher struct {
```

Comment:
I would name this `RegexStringsWatcher` to indicate it is a variation on the `StringsWatcher` theme.